docs/proposals/20220105-observatorium-alerting.md

```diff
@@ -0,0 +1,124 @@
+# Observatorium Alerting
+
+* **Owners:**
+  * `@onprem` `@bwplotka`
+
+* **Related Tickets:**
+  * https://issues.redhat.com/browse/MON-1761
+
+* **Other docs:*
+  * -????
+
+> TL;DR: For monitoring use cases, especially tenants of the metric signal of Observatorium has to be able to configure Prometheus-based alerts that will be evaluated for a given interval. If triggered they should notify desired target (receiver) e.g PagerDuty using Alertmanager Configuration. Users should also be able to see the status of all alerts through Prometheus-like Alerts HTTP API. This proposal specifies how such a system can be deployed within Observatorium.
+
+## Why
+
+Alerting is a critical component of reliable monitoring. We can build automation or notify humans in a reactive or proactive manner instead of watching dashboards.
+
+### Pitfalls of the current solution
+
+Observatorium did not have alerting defined before.
+
+## Goals
+
+Definition: 
+* Alert Routing: Configuration of Alertmanager related to receiver target config and its routing. 
+
+Goals:
+* Tenant should be able to use alerting rules.
+  * Alerting Rules via GET and PUT via HTTP (Rules API)
+* Tenants should be able to read what is the status of configured alerts (Prometheus Alert API GET)
+* Tenants are able to route alerts:
+  * Alerting Routing via GET and PUT via HTTP API.
+* Both alerts and alerting rules should be multi-tenant.
+
+## Non-Goals
+
+* (?) UI for Alerts/Rules, Alert Status, AM Routing, AM Alerts …
+* (?) Alertmanager Alerts API, Config/Status
+* Silences
+* Inhibits
+* Scaling Rulers
+* Scaling Alertmanager beyond 3 replicas.
+* Routing tester.
+
+## Audience of this proposal
+
+Observatorium Devs and Users.
+
+## How
+
+Within work for [Rules API proposal](https://observatorium.io/docs/proposals/20201019-prometheus-rules-in-observatorium-api.md/) we already added Rules APIs for reading and configuring rules which include both recording and alerting rules.
+
+For example, to save alerts we can already PUT the following object to `/api/metrics/v1/{tenant}/api/v1/rules/raw`
+
+```
+groups:
+  - name: test-oidc
+    interval: 5s
+    rules:
+      - alert: HighRequestLatency
+        expr: job:request_latency_seconds:mean5m{job="second"} > 0.5
+        for: 10m
+        labels:
+          team: alpha
+```
+
+Such rule will be saved through Observatorium API, then Rule-Store to object storage for the specified tenant. It can be then listed through a similar endpoint and GET.
+
+### Alerts
+
+Such configuration will be then eventually synchronized with tenant’s Rulers using ConfigSync. The Ruler itself has to communicate on 4 levels. All connections are presented in the diagram below.
+
+![ruler-connections-diagram](../assets/alerting-ruler.png)
+
+1. Alerts will be evaluated based on periodic requests to any replica of  HTTP Query API of Query-frontend (Question: Or querier?). 
```

Comment:
> (Question: Or querier?).

Why not directly through Query FE?

Reply:
> Why not directly through Query FE?

What @matej-g is saying makes sense. It's an implementation detail we can figure out later.

Reply:
All addressed, except this:

> My understanding is that when the Rules API is active, alerts will already be evaluated in Thanos Ruler? The outcome is that Ruler will create new rules with __name__ = ALERT?

@ianbillett could you elaborate on what do you mean? Outcome is that Recording Rule is evaluated and it's output is pushed to  receiver. Similarly, Alert is evaluated and if triggered it is sending `__name__ = ALERT?` and pushing Alert to AM.

Reply:
I guess what I mean here is that - when the Rules API is implemented, we don't need to do anything else in this Alerting proposal, because the Ruler is responsible for evaluating the alerts and sending to Alertmanager. All we need to do is tell Ruler where to forward the alerts.